packages/integration-tests/src/test-e2e/hosting.test.ts

```diff
@@ -0,0 +1,351 @@
+  let executionRoleArn: string;
+
+  before(async () => {
+    assert.ok(
+      process.env.AMPLIFY_BACKEND_TESTS_HOSTING_TEST_BRANCH_NAME,
+      'AMPLIFY_BACKEND_TESTS_HOSTING_TEST_BRANCH_NAME environment variable must be set.',
+    );
+    branchName = process.env.AMPLIFY_BACKEND_TESTS_HOSTING_TEST_BRANCH_NAME;
+    assert.ok(
+      process.env.AMPLIFY_BACKEND_TESTS_HOSTING_TEST_COMMIT_SHA,
+      'AMPLIFY_BACKEND_TESTS_HOSTING_TEST_COMMIT_SHA environment variable must be set.',
+    );
+    commitSha = process.env.AMPLIFY_BACKEND_TESTS_HOSTING_TEST_COMMIT_SHA;
+    assert.ok(
+      process.env.AMPLIFY_BACKEND_TESTS_E2E_EXECUTION_ROLE_ARN,
+      'AMPLIFY_BACKEND_TESTS_E2E_EXECUTION_ROLE_ARN environment variable must be set.',
+    );
+    executionRoleArn = process.env.AMPLIFY_BACKEND_TESTS_E2E_EXECUTION_ROLE_ARN;
+    appId = await findTestingAppId(executionRoleArn);
+    await pruneStaleBranches(appId);
+    await ensureBranchIsConnected(appId, branchName);
+  });
+  void it('can deploy backend', async () => {
+    const deploymentJob = await startOrGetDeploymentJob(
+      appId,
+      branchName,
+      commitSha,
+    );
+    if (deploymentJob.status !== JobStatus.SUCCEED) {
+      assert.ok(deploymentJob.jobId);
+      await waitForSuccessfulJobCompletion(
+        appId,
+        branchName,
+        deploymentJob.jobId,
+      );
+    }
+  });
+});
+
+const waitForSuccessfulJobCompletion = async (
+  appId: string,
+  branchName: string,
+  jobId: string,
+) => {
+  const timeoutMs = 10 * 60 * 1000; // 10 minutes;
+  const pollingInterval = 10 * 1000; // 10 seconds;
+  const startTime = Date.now();
+  while (Date.now() - startTime <= timeoutMs) {
+    const getJobResult = await amplifyClient.send(
+      new GetJobCommand({
+        appId,
+        branchName,
+        jobId,
+      }),
+    );
+    const jobSummary = getJobResult.job?.summary;
+    assert.ok(jobSummary);
+    console.log(`Job ${jobId} is in status ${jobSummary.status}`);
+    // Fail if job is in terminal non-successful state.
+    assert.ok(
+      jobSummary.status !== 'FAILED',
+      `Job ${jobId} has failed. Check logs in the AWS Console.`,
+    );
+    assert.ok(
+      jobSummary.status !== 'CANCELLED',
+      `Job ${jobId} has been canceled. Check logs in the AWS Console.`,
+    );
+    if (jobSummary.status === 'SUCCEED') {
+      return;
+    }
+    await new Promise((resolve) => setTimeout(resolve, pollingInterval));
+  }
+};
+
+const startOrGetDeploymentJob = async (
+  appId: string,
+  branchName: string,
+  commitSha: string,
+): Promise<JobSummary> => {
+  return runWithRetry(
+    async () => {
+      const existingJobs = await listJobs(appId, branchName);
+      // Find existing job that's either successful or is in progress first that belongs to the commit.
+      const existingJob = existingJobs.find((item) => {
+        return (
+          item.commitId === commitSha &&
+          (item.status === JobStatus.SUCCEED ||
```

Comment:
Will rerunning an e2e workflow skip the creation of a new job? There seems to be a [DeleteJob](https://docs.aws.amazon.com/amplify/latest/APIReference/API_DeleteJob.html) API that we might be able to use for cleaning up.